pkg/coscheduling/scheduler/core/core.go

```diff
@@ -0,0 +1,533 @@
+		sumInfo[pgName] = GroupInfo{
+			resource: r,
+			progress: float32(pgs.PodGroup.Status.Scheduled) / float32(pgs.PodGroup.Spec.MinMember),
+		}
+	}
+
+	if len(sumInfo) == 0 {
+		return nil, false
+	}
+
+	klog.V(6).Infof("Potential eviction pod group list %+v", sumInfo)
+	if !checkLeftResourceSatisfy(framework.NewResource(sum.ResourceList()), framework.NewResource(*current.PodGroup.Spec.MinResources)) {
+		klog.Warning("No potential eviction pod groups")
+		return nil, false
+	}
+	simulatorEvictionList := make([]ResourcePair, 0)
+	list := sortMapByValue(sumInfo)
+	// add the pod group resource of current pod
+	clusterLeft.Add(sumInfo[pgFullName].resource.ResourceList())
+	for _, resourcePair := range list {
+		if resourcePair.resource == nil {
+			continue
+		}
+		// exclude the pod group resource of current pod, current pg should not evict
+		if resourcePair.pgName == pgFullName {
+			continue
+		}
+		clusterLeft.Add(resourcePair.resource.ResourceList())
+		simulatorEvictionList = append(simulatorEvictionList, resourcePair)
+		if checkLeftResourceSatisfy(clusterLeft, framework.NewResource(*current.PodGroup.Spec.MinResources)) {
+			return simulatorEvictionList, true
+		}
+	}
+	return nil, false
+}
+
+// evictGroup do eviction of pod group
+func (sop *ScheduleOperation) evictGroup(evictList []ResourcePair) string {
+	// nodeName if used for preemption(PostFilter), we just need to return anyone
+	nodeName := ""
+	for _, rp := range evictList {
+		klog.Infof("Evict group %v", rp.pgName)
+		pgs := sop.podGroupStatusCache.Get(rp.pgName)
+		if len(nodeName) == 0 {
+			obj, ok := pgs.MatchedPodNodes.Get(string(pgs.Pod.UID))
+			if !ok {
+				continue
+			}
+			nodeName = obj.(*cache.PodNodePair).Node
+
+		}
+		pgs.MatchedPodNodes.Delete(string(pgs.Pod.UID))
+		pgs.PodNameUIDs.Delete(fmt.Sprintf("%v/%v", pgs.Pod.Namespace, pgs.Pod.Name))
+		sop.AddToDenyCache(rp.pgName)
+	}
+	return nodeName
+}
+
+// fillOccupiedObj fill some infomation to pod group cache
+func (sop *ScheduleOperation) fillOccupiedObj(pgs *cache.PodGroupSchedulingStatus, pod *corev1.Pod) error {
+	if pgs == nil || pgs.PodGroup == nil {
+		return fmt.Errorf("PodGroupSchedulingStatus is nil")
+	}
+	pgFullName := fmt.Sprintf("%v/%v", pod.Namespace, pgs.PodGroup.Name)
+	refs := make([]string, 0)
+	for _, ownerRef := range pod.OwnerReferences {
+		refs = append(refs, string(ownerRef.UID))
+	}
+	if pgs.Pod == nil {
+		pgs.Pod = pod
+	}
+	if pgs.PodGroup.Spec.MinResources == nil {
+		// initialize pod group resource
+		req := getPodGroupMinResourceFromPod(pod, int(pgs.PodGroup.Spec.MinMember)).ResourceList()
+		pgs.PodGroup.Spec.MinResources = &req
+	}
+	if pgs.PodGroup.Status.OccupiedBy == "" {
+		// if do not have refs or initialize, return nil
+		if len(refs) != 0 {
+			// initialize pod group
+			sortkeys.Strings(refs)
+			pgs.PodGroup.Status.OccupiedBy = strings.Join(refs, ",")
+		}
+		return nil
+	}
+	// if refs not match, return error
+	if len(refs) == 0 {
```

Comment:
I think you should move this to a place which can return as early as possible, for instance L438?

Reply:
yep, thanks.

Reply:
sorry, we can not do this.

We check `pgs.PodGroup.Status.OccupiedBy` is not nil, but len(refs)==nil, that means pod was created by anther workload, we we should return error.